### DISCUSSION TITLE
Make Incremental Static Regeneration route type agnostic

### DISCUSSION THREAD

**Discussion by stefanhar** [vercel/next.js#29258] [Ideas]
### Describe the feature you'd like to request

I believe we have a case where using ISR on static paths would be useful.
We want to limit the effects external dependencies have on our build pipelines so we get page content only at runtime.

It seems that currently the best way for us to do this is to use SSR, and depend on external caching methods to cache the pages.

It would be nice to have something like ISR for static path, this would allow us handle cache within our project simply via Next JS.

It is probably possible to work around this by catching all routes as dynamic routes in the root and having a root page component that render all other page types, but that seems a bit hacky.

### Describe the solution you'd like

Have `renderBehaviour` option in static route pages.
`build`: Page is created on build time (default)
`fallback`: Page is created when requested and returns a `Fallback page`
`blocking`: Page is created when requested (SSR) and is returned when ready (blocking)

### Describe alternatives you've considered

- Move `fallback`option into `getStaticProps` function.
A static route page with `fallback` would follow rules dictated in fallback 
- Make no distinction between static and dynamic paths allow use of `getStaticProps` and `getStaticPaths` on all pages

**Comment by michaelschufi** [comment]
Yes, this is what our team is looking for as well.

We have dynamic content that is the same for all users and using ISR is perfect. However, during build we don't know yet what content there is or if it will be outdated by the time the build gets released.

**Comment by Skandalabrandur** [comment]
This is something I struggle with as well
